tests/integration/crunchers/test_cruncher_equal_quantile_walk.py

```diff
@@ -0,0 +1,351 @@
+            [_mb, _sc, "World", _ech4, _mtch4, 220],
+            [_mb, _sd, "World", _ech4, _mtch4, 50],
+            [_mb, _se, "World", _ech4, _mtch4, 150],
+        ],
+        columns=_msrvu + [2010],
+    )
+
+    small_db = pd.DataFrame(
+        [[_mb, _sa, "World", _eco2, _gtc, 1.2], [_ma, _sb, "World", _eco2, _gtc, 2.3]],
+        columns=_msrvu + [2010],
+    )
+
+    tdownscale_df = pd.DataFrame(
+        [
+            [_mc, _sa, "World", _eco2, _gtc, 1, 2, 3, 4],
+            [_mc, _sb, "World", _eco2, _gtc, 0.5, 0.5, 0.5, 0.5],
+            [_mc, _sc, "World", _eco2, _gtc, 5, 5, 5, 5],
+            [_ma, _sc, "World", _eco2, _gtc, 1.5, 2.5, 2.8, 1.8],
+        ],
+        columns=_msrvu + [2010, 2030, 2050, 2070],
+    )
+
+    simple_df = pd.DataFrame(
+        [
+            [_mc, _sa, "World", _eco2, _gtc, 0, 1000, 5000],
+            [_mc, _sb, "World", _eco2, _gtc, 1, 1000, 5000],
+            [_mc, _sa, "World", _ech4, _mtch4, 0, 300, 500],
+            [_mc, _sb, "World", _ech4, _mtch4, 1, 300, 500],
+        ],
+        columns=_msrvu + [2010, 2030, 2050],
+    )
+
+    def test_derive_relationship(self, test_db):
+        tcruncher = self.tclass(test_db)
+        res = tcruncher.derive_relationship("Emissions|CO2", ["Emissions|CH4"])
+        assert callable(res)
+
+    def test_derive_relationship_with_nans(self):
+        tdb = self.tdb.copy()
+        tdb.loc[(tdb["variable"] == _eco2) & (tdb["model"] == _ma), 2050] = np.nan
+        tcruncher = self.tclass(IamDataFrame(tdb))
+        res = tcruncher.derive_relationship("Emissions|CO2", ["Emissions|CH4"])
+        # just make sure that this runs through and no error is raised
+        assert callable(res)
+
+    def test_derive_relationship_with_multicolumns(self):
+        tdb = self.tdb.copy()
+        tcruncher = self.tclass(IamDataFrame(tdb))
+        error_msg = re.escape(
+            "For `EqualQuantileWalk`, ``variable_leaders`` should only "
+            "contain one variable"
+        )
+        with pytest.raises(ValueError, match=error_msg):
+            tcruncher.derive_relationship(
+                "Emissions|CO2", ["Emissions|CH4", "Emissions|HFC|C5F12"],
+            )
+
+    @pytest.mark.parametrize("add_col", [None, "extra_col"])
+    def test_relationship_usage(self, test_db, simple_df, add_col):
+        tcruncher = self.tclass(test_db)
+        lead = ["Emissions|CO2"]
+        follow = "Emissions|CH4"
+        simple_df = self._adjust_time_style_to_match(simple_df, test_db)
+        res = tcruncher.derive_relationship(follow, lead)
+        if add_col:
+            add_col_val = "blah"
+            simple_df[add_col] = add_col_val
+            simple_df = IamDataFrame(simple_df.data)
+            assert simple_df.extra_cols[0] == add_col
+
+        infilled = res(simple_df)
+        # We compare the results with the expected results: for T1, we are below the
+        # lower limit on the first, in the middle on the second. At later times we are
```

Comment:
```suggestion
        # lower limit on the first, in the middle on the second scenario. At later times we are
```

? Is there a way to make it slightly clearer in the test rather than having the hard-coded 50 and 100?

Reply:
It's derived better now